PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs

```diff
@@ -0,0 +1,390 @@
+        {
+            Mock<Microsoft.VisualStudio.Text.Operations.IEditorOperations> editorOperations = new Mock<IEditorOperations>();
+            editorOperations.Setup(m => m.AddBeforeTextBufferChangePrimitive())
+                            .Callback(() => { });
+            editorOperations.Setup(m => m.AddAfterTextBufferChangePrimitive())
+                            .Callback(() => { });
+
+            // {|} --> {p|}
+            editorOperations.Setup(m => m.InsertText(It.IsAny<string>()))
+                            .Callback<string>(p =>
+                                                {
+                                                    _mockedScript = _mockedScript.Insert(_mockedCaret, p);
+                                                    _mockedCaret++;
+                                                });
+
+            // {|} --> {|
+            editorOperations.Setup(m => m.Delete())
+                            .Callback(() =>
+                                        {
+                                            _mockedScript = _mockedScript.Remove(_mockedCaret, 1);
+                                        });
+
+            // {}| --> {|}  
+            editorOperations.Setup(m => m.MoveToPreviousCharacter(false))
+                            .Callback(() =>
+                                        {
+                                            _mockedCaret--;
+                                        });
+
+            // {|} --> {}|
+            editorOperations.Setup(m => m.MoveToNextCharacter(false))
+                            .Callback(() =>
+                                        {
+                                            _mockedCaret++;
+                                        });
+
+            // {|} --> {\r\n|}
+            editorOperations.Setup(m => m.InsertNewLine())
+                            .Callback(() =>
+                                        {
+                                            _mockedScript = _mockedScript.Insert(_mockedCaret, Environment.NewLine);
+                                            _mockedCaret += Environment.NewLine.Length;
+                                        });
+
+            // {\r\n|} --> {|\r\n}
+            editorOperations.Setup(m => m.MoveLineUp(false))
+                            .Callback(() =>
+                                        {
+                                            _mockedCaret = _mockedScript.Substring(0, _mockedCaret).LastIndexOf(Environment.NewLine);
+                                        });
+
+            editorOperations.Setup(m => m.MoveToEndOfLine(false))
+                            .Callback(() =>
+                                        {
+                                            // Should move caret to the end, here we remain it still as it's already at the end.
+                                        });
+
+            // {\r\n|\r\n} --> {\r\n  |\r\n}
+            editorOperations.Setup(m => m.Indent())

```

Comment:
Yes. Besides this, we'll need to add some logic here when we have nested braces/quotes supported.